qiita_db/sql_connection.py

```diff
@@ -22,57 +22,139 @@
+
+* Querying
+
+In order to perform any query in the database you first need to import the
+`TRN` variable available in this module:
+
+>>> from qiita_db.sql_connection import TRN
+
+The `TRN` variable is an instance of the Transaction object. All queries should
+be executed through this transaction object to avoid committing anything to the
+database that is partially executed. To perform any query you first need to
+add the query to the transaction and then execute the transaction:
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     res = TRN.execute()
+>>> res
+[[[42]]]
+
+* Data retrieval
+
+Multiple auxiliary functions exist for easy SQL result retrieval:
+
+Getting the first value of the last SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchlast()
+>>> res
+42
+
+Getting the results of the last SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchindex()
+>>> res
+[[42]]
+
+Getting the results of the specified SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 43")
+...     res = TRN.execute_fetchindex(0)
+>>> res
+[[42]]
+
+Getting the results of the last SQL query flattened
+>>> with TRN:
+...     TRN.add("SELECT 42, 43, 44")
+...     res = TRN.execute_fetchflatten()
+>>> res
+[42, 43, 44]
+
+Getting the results of the specified SQL query flattened
+>>> with TRN:
+...     TRN.add("SELECT 42, 43, 44")
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchflatten(0)
+>>> res
+[42, 43, 44]
+
+* Transactions
+
+Transaction blocks are created through the same `TRN` variable exported in this
+module. You can add as many SQL commands as you want and execute all of them at
+once, and it will return the results of all the SQL commands. `TRN` should be
+used as a context so manager, and it autocommits the transaction once the last
+context is exited, as long as no error was generated inside the context, in
+which case a rollback is executed.
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 43")
+...     res = TRN.execute()  # The transaction is not committed here
+>>> # The transaction committed here
+>>> res
+[[[42]], [[43]]]
+
+You can have nested transactions and they will not commit until the first
+transaction (represented by entering to the context) is exited:
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     with TRN:
+...         TRN.add("SELECT 43")
+...         res = TRN.execute()
+...     # The transaction is still not committed
+...     TRN.add("SELECT 44")
+...     res = TRN.execute()
+>>> # The transactions committed here
+>>> res
+[[[42]], [[43]], [[44]]]
+
+You can also use results from a previous command in the transaction in a later
+command without partially executing the transaction by using placeholders. The
+placeholders are of the form {int:int:int}, where the first integer is the
+index of the query in the transaction, the second integer is the row in the
```

Comment:
The row is difficult to understand since the cursor is not assured (AFAIK) to return rows in any particular order. In other words, the database is unordered unless a requested order is specified. I can see how this works when a single row is returned, but I can't help but thing anything besides `0` is meaningful unless the query includes an explicit ordering AND the developer knows in advance what the returned row of interest will be